go/pkg/sysdb/grpc/collection_service_test.go

```diff
@@ -572,6 +573,39 @@ func (suite *CollectionServiceTestSuite) TestGetCollectionSize() {
 	suite.NoError(err)
 }
 
+func (suite *CollectionServiceTestSuite) TestCountForks() {
+	collectionName := "collection_service_test_count_forks"
+	collectionID, err := dao.CreateTestCollection(suite.db, collectionName, 128, suite.databaseId)
+	suite.NoError(err)
+
+	req := coordinatorpb.CountForksRequest{
+		SourceCollectionId: collectionID,
+	}
+	res, err := suite.s.CountForks(context.Background(), &req)
+	suite.NoError(err)
+	suite.Equal(uint64(0), res.Count)
+
+	// Create 5 forks
+	for i := 0; i < 5; i++ {
+		forkCollectionReq := &coordinatorpb.ForkCollectionRequest{
+			SourceCollectionId:                   collectionID,
+			SourceCollectionLogCompactionOffset:  0,
+			SourceCollectionLogEnumerationOffset: 0,
+			TargetCollectionId:                   types.NewUniqueID().String(),
+			TargetCollectionName:                 fmt.Sprintf("test_fork_collection_fork_%d", i),
+		}
+		_, err = suite.s.ForkCollection(context.Background(), forkCollectionReq)
+		suite.NoError(err)
+	}
+
+	res, err = suite.s.CountForks(context.Background(), &req)
```

Comment:
wondering if this should be called `CalculateForkTreeSize` or similar
also might be good to test here that the result is 5 for `CountForks` on every collection in the set